### PR TITLE
docs: basic documentation

### LINKED ISSUE #3: Update initial docs
**What to document**

- [ ] README
- [ ] Release process

### PR DESCRIPTION
**Pull request by therealbobo**
Signed-off-by: Roberto Scolaro [roberto.scolaro21@gmail.com](roberto.scolaro21@gmail.com)

**What type of PR is this?**

/kind documentation

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

/area documentation

**What this PR does / why we need it**:

Adding basic documentation to the repository.

**Which issue(s) this PR fixes**:

Fixes #3 

**Special notes for your reviewer**:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,61 @@
...
+### Registering a new Rule
+
+Registering your rule inside the registry helps ensure that some technical constraints are respected. Moreover, this is a great way to share your rule project with the community and engage with it, thus gaining new users and **increasing its visibility**. We encourage you to register your rule in this registry before publishing it.
+
+The registration process involves adding an entry about your rule inside the [registry.yaml](./registry.yaml) file by creating a Pull Request in this repository. Please be mindful of a few constraints that are automatically checked and required for your rule to be accepted:
+
+- The `name` field is mandatory and must be **unique** across all the rule in the registry
+- The rule `name` must match this [regular expression](https://en.wikipedia.org/wiki/Regular_expression): `^[a-z]+[a-z0-9-_\-]*$` (however, its not reccomended to use `_` in the name, unless you are trying to match the name of a source or for particular reasons)
+- The `path` field should specify the path to the rule in this repository
+- The `url` field should point to the plugin source code
```
Comment (jasondellaluce) [inline README.md:24]:
```suggestion
- The `url` field should point to the ruleset file in the source code
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+2. Please label this pull request according to what type of issue you are addressing.
+3. Please add a release note!
+4. If the PR is unfinished while opening it specify a wip in the title before the actual title, for example, "wip: my awesome feature"
+-->
+
+**What type of PR is this?**
+
+> Uncomment one (or more) `/kind <>` lines:
+
+> /kind bug
```
Comment (jasondellaluce) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: **todo:** also update [test-infra's config](https://github.com/falcosecurity/test-infra/blob/d8925b88d0ba19d40ad5167a5c1397a95019b8aa/config/plugins.yaml#L130) for labels in this repo.

File: README.md [hunk 3]
```diff
@@ -1,9 +1,61 @@
...
+
+This repository contains the [Rule Registry](#rule-registry) and the [rules officially maintained](#falcusecurity-rules) by the Falcosecurity organization. [Rules](https://falco.org/docs/rules) tell [Falco](https://github.com/falcosecurity/falco) what to do. Please refer to the [official documentation](https://falco.org/docs/rules) to better understand the rules' concepts. 
+
+## Rule Registry
+
+The Registry contains metadata and information about every rule known and recognized by the Falcosecurity organization. These rules are developed for Falco and made available to the community. Check out the sections below to know how to [register your rules](#registering-a-new-rule) and see rules currently contained in the registry.
+
+### Registering a new Rule
+
+Registering your rule inside the registry helps ensure that some technical constraints are respected. Moreover, this is a great way to share your rule project with the community and engage with it, thus gaining new users and **increasing its visibility**. We encourage you to register your rule in this registry before publishing it.
```
Comment (jasondellaluce) [inline README.md:17]:
```suggestion
Registering your rule inside the registry helps ensure that some technical constraints are respected. Moreover, this is a great way to share your ruleset and make it available to the community. We encourage you to register your ruleset in this registry before publishing it.
```

File: README.md [hunk 4]
```diff
@@ -1,9 +1,61 @@
...
+
+The registration process involves adding an entry about your rule inside the [registry.yaml](./registry.yaml) file by creating a Pull Request in this repository. Please be mindful of a few constraints that are automatically checked and required for your rule to be accepted:
+
+- The `name` field is mandatory and must be **unique** across all the rule in the registry
+- The rule `name` must match this [regular expression](https://en.wikipedia.org/wiki/Regular_expression): `^[a-z]+[a-z0-9-_\-]*$` (however, its not reccomended to use `_` in the name, unless you are trying to match the name of a source or for particular reasons)
```
Comment (jasondellaluce) [inline README.md:22]:
```suggestion
- The rule `name` must match this [regular expression](https://en.wikipedia.org/wiki/Regular_expression): `^[a-z]+[a-z0-9-_\-]*$` (however, its not reccomended to use `_` in the name)
```

File: README.md [hunk 5]
```diff
@@ -1,9 +1,61 @@
...
+
+### Registered Rules
+
+Please refer to the automatically generated [rules_inventory/rules_overview.md](https://github.com/falcosecurity/rules/blob/main/rules_inventory/rules_overview.md#falco-rules---detailed-overview) file for a detailed list of all the rules currently registered.
+
+## Hosted Rules
+
+Another purpose of this repository is to host and maintain the rules owned by the Falcosecurity organization. All the rules are contained inside the [rules](https://github.com/falcosecurity/rules/tree/main/rules) folder.
+
+The `master` branch contains the most up-to-date state of development. Please check our [Release Process](./RELEASE.md) to know how rules are released. Dev builds are published each time a Pull Request gets merged into `master`, whereas stable builds are released and published only when a new release gets tagged.
```
Comment (jasondellaluce) [inline README.md:50]:
```suggestion
The `main` branch contains the most up-to-date state of development. Please check our [Release Process](./RELEASE.md) to know how rules are released. Stable builds are released and published only when a new release gets tagged.
```

File: README.md [hunk 6]
```diff
@@ -1,9 +1,61 @@
...
+
+If you wish to contribute your plugin to the Falcosecurity organization, you just need to open a Pull Request to add it inside the `rules` folder and to add it inside the registry. In order to be hosted in this repository, plugins must be licensed under the [Apache 2.0 License](./LICENSE).
```
Comment (jasondellaluce) [inline README.md:52]:
```suggestion
If you wish to contribute your rules to the Falcosecurity organization, you just need to open a Pull Request to add it inside the `rules` folder and to add it inside the registry. In order to be hosted in this repository, plugins must be licensed under the [Apache 2.0 License](./LICENSE). 
```

File: README.md (hunk 3 above)
Comment (leogr) [reply README.md:17]: We don't know yet if this repository is open to any community-curated ruleset or if it's reserved only for Falco official rules files. So I'd avoid setting wrong expectations at the moment.

File: README.md [hunk 7]
```diff
@@ -1,9 +1,61 @@
...
-## ⚠️ Current status
+Note: *This repository has been created upon this [proposal](https://github.com/falcosecurity/falco/blob/master/proposals/20221129-artifacts-distribution.md#move-falco-rules-to-their-own-repo) and is not yet ready for use. The progress tracker can be found [here](https://github.com/falcosecurity/evolution/issues/229).*
+
+This repository contains the [Rule Registry](#rule-registry) and the [rules officially maintained](#falcusecurity-rules) by the Falcosecurity organization. [Rules](https://falco.org/docs/rules) tell [Falco](https://github.com/falcosecurity/falco) what to do. Please refer to the [official documentation](https://falco.org/docs/rules) to better understand the rules' concepts. 
+
+## Rule Registry
+
+The Registry contains metadata and information about every rule known and recognized by the Falcosecurity organization. These rules are developed for Falco and made available to the community. Check out the sections below to know how to [register your rules](#registering-a-new-rule) and see rules currently contained in the registry.
+
+### Registering a new Rule
```
Comment (leogr) [inline README.md:15]:
```suggestion
### Registering a new Rules file
```

Technically, the smallest working unit for this repository is a _rules file_ and not a _rule_.
This should be reflected in the document.

**Review by leogr — COMMENTED** [review]
This is awesome! :clap: 

Just left a couple of comments.

File: README.md (hunk 6 above)
Comment (LucaGuerra) [inline README.md:52]:
```suggestion
If you wish to contribute your rulesfile to the Falcosecurity organization, you just need to open a Pull Request to add it inside the `rules` folder and to add it inside the registry. In order to be hosted in this repository, rulesfiles must be licensed under the [Apache 2.0 License](./LICENSE). 
```

File: README.md (hunk 3 above)
Comment (LucaGuerra) [reply README.md:17]: Which means that we can leave out the entire "registering a new rule" section for the moment but leave the registry format as documentation?

**Comment by therealbobo** [comment]
I updated the PR according to your feedback! 😄

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (therealbobo) [reply .github/PULL_REQUEST_TEMPLATE.md:13]: Done! ✔️ [https://github.com/falcosecurity/test-infra/pull/971](https://github.com/falcosecurity/test-infra/pull/971)

**Review by jasondellaluce — APPROVED** [review]
/approve

Thanks!

**Pull request merged**